<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ExpressionsDialog</class>
 <widget class="QDialog" name="ExpressionsDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>479</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Compose expression</string>
  </property>
  <property name="sizeGripEnabled">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1">
   <property name="sizeConstraint">
    <enum>QLayout::SetFixedSize</enum>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>Compose Expression</string>
         </property>
         <layout class="QGridLayout" name="gridLayout" columnstretch="0,1,0">
          <item row="4" column="1">
           <widget class="QLabel" name="lblOut2">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="lblOut0">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Input</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Output</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLineEdit" name="lineIn1"/>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="lblOut1">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLineEdit" name="lineIn2"/>
          </item>
          <item row="2" column="0">
           <widget class="QLineEdit" name="lineIn0"/>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QLineEdit" name="lineExpression">
            <property name="font">
             <font>
              <family>Ubuntu Mono</family>
              <pointsize>12</pointsize>
             </font>
            </property>
            <property name="text">
             <string>VAL</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Expression</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <property name="buddy">
             <cstring>lineExpression</cstring>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btnAccept">
         <property name="text">
          <string>&amp;Accept</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnCancel">
         <property name="text">
          <string>&amp;Cancel</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnInfo">
         <property name="text">
          <string>&amp;Show info</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QWidget" name="widgetInfo" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QLabel" name="lblInfo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:12pt; font-weight:600;&quot;&gt;Expression Information&lt;/span&gt;&lt;/p&gt;&lt;p&gt;The value read via modbus is represented as &amp;quot;VAL&amp;quot;. The most common binary operators are supported (!, |, &amp;amp;, &amp;lt;&amp;lt;, &amp;gt;&amp;gt;). The basic arithmetic operators are also supported (+, -, *, /, %, ^). Hexadecimal numbers can be represented with the &amp;quot;0x&amp;quot; prefix. Floating point number are also supported. Both a decimal point as comma can be used. The first encountered characters per expression is used as floating point separator.&lt;/p&gt;&lt;p&gt;Some examples:&lt;/p&gt;&lt;p&gt;(VAL + 10 / 2) / 10&lt;/p&gt;&lt;p&gt;(VAL &amp;amp; 0xFF) | 1&lt;/p&gt;&lt;p&gt;VAL * 0.001&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>lineExpression</tabstop>
  <tabstop>lineIn0</tabstop>
  <tabstop>lineIn1</tabstop>
  <tabstop>lineIn2</tabstop>
  <tabstop>btnAccept</tabstop>
  <tabstop>btnCancel</tabstop>
  <tabstop>btnInfo</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
